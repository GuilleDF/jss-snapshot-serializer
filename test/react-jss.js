import React from 'react';
import { createUseStyles } from 'react-jss';
import { render } from '@testing-library/react';
import serializer from '../src/jss-snapshot-serializer';

expect.addSnapshotSerializer(serializer);

const useStyles = createUseStyles({
  redBox: {
    width: 100,
    height: 100,
    backgroundColor: 'red',
  },
});

const RedBox = () => {
  const classes = useStyles();
  return <div className={classes.redBox} />;
};

describe('react-jss', () => {
  it('removes non-deterministic class names', () => {
    const redBox = render(<RedBox />).asFragment();
    expect(redBox).toMatchSnapshot();
  });
});
