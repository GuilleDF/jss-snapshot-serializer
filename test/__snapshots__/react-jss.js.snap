// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-jss removes non-deterministic class names 1`] = `
<DocumentFragment>
  <div
    class="redBox-0-2-1"
  />
</DocumentFragment>
`;
